src/node.cc

```diff
@@ -452,9 +456,24 @@ void LoadEnvironment(Environment* env) {
 
 
 #ifdef __POSIX__
+
+struct sigaction previous_sigsegv_action;
+
+void TrapWebAssemblyOrContinue(int sig, siginfo_t *info, void *ucontext) {
+  if (!v8::TryHandleWebAssemblyTrapPosix(sig, info, ucontext)) {
+    if (previous_sigsegv_action.sa_handler) {
+      previous_sigsegv_action.sa_handler(sig);
+    }
+  }
+}
+
 void RegisterSignalHandler(int signal,
                            void (*handler)(int signal),
                            bool reset_handler) {
+  if (signal == SIGSEGV) {
+    previous_sigsegv_action.sa_handler = handler;
+    return;
+  }
```

Comment:
Is the idea to expose `RegisterSignalHandler()` as a public API? Joyee already pointed out the signature of the signal handler.

I think another issue is that users should be able to expect that chaining handlers Just Works. Not just for one specific signal and not just one level deep.

Aside: you should use atomic ops for loads and stores because otherwise the compiler is technically allowed to reorder the store with the sigaction() call.